internal/oc/exporter.go

```diff
@@ -0,0 +1,61 @@
+package oc
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"go.opencensus.io/trace"
+)
+
+var _ = (trace.Exporter)(&LogrusExporter{})
+
+// LogrusExporter is an OpenCensus `trace.Exporter` that exports
+// `trace.SpanData` to logrus output.
+type LogrusExporter struct {
+}
+
+// ExportSpan exports `s` based on the the following rules:
+//
+// 1. All output will contain `s.Attributes`, `s.TraceID`, `s.SpanID`,
+// `s.ParentSpanID` for correlation
+//
+// 2. All `Annotations` will be written previous to the span itself.
+//
+// 3. Annotations will combine #1 with the annotation specific attributes and be
+// written at `logrus.DebugLevel` unless the `Annotation.Attributes` contains a
+// key `logrus.ErrorKey`.
+//
+// 4. The span itself will be written at `logrus.InfoLevel` unless
+// `s.Status.Code != 0` in which case it will be written at `logrus.ErrorLevel`
+// providing `s.Status.Message` as the error value.
+func (le *LogrusExporter) ExportSpan(s *trace.SpanData) {
+	// Combine all span annotations with traceID, spanID, parentSpanID
+	baseEntry := logrus.WithFields(logrus.Fields(s.Attributes))
+	baseEntry.Data["traceID"] = s.TraceID.String()
+	baseEntry.Data["spanID"] = s.SpanID.String()
+	baseEntry.Data["parentSpanID"] = s.ParentSpanID.String()
+
+	// Log all Annotations combined with Annotation attributes
+	for _, a := range s.Annotations {
+		aEntry := baseEntry.WithFields(a.Attributes)
+		level := logrus.DebugLevel
+		if _, ok := aEntry.Data[logrus.ErrorKey]; ok {
+			level = logrus.ErrorLevel
+		}
+		aEntry.Time = a.Time
```

Comment:
I don't think we include the Time stored in the Logrus entry in the ETW hook, so I'd recommend logging this as a separate field like `annotationTime`.

Reply:
Why? When this is written out the logrus entry will be "some time after" the annotation actually took place. Why not just have logrus write it out as if it happened at the time the annotation actually occurred?